packages/venia-concept/src/components/Checkout/wrapper.js

```diff
@@ -16,53 +15,51 @@ import {
 
 import Flow from './flow';
 
-const { BrowserPersistence } = Util;
-const storage = new BrowserPersistence();
-
-const isAddressValid = address => !!(address && address.email);
+const hasData = value => !!value;
 const isCartReady = cart => cart.details.items_count > 0;
 const isCheckoutReady = checkout => {
-    return (
-        isPaymentMethodReady() &&
-        isShippingInfoReady() &&
-        isShippingMethodReady(checkout)
+    const {
+        billingAddress,
+        paymentData,
+        shippingAddress,
+        shippingMethod
+    } = checkout;
+
+    return [billingAddress, paymentData, shippingAddress, shippingMethod].every(
+        hasData
     );
 };
-const isPaymentMethodReady = () => {
-    const paymentMethod = storage.getItem('paymentMethod');
-    return !!paymentMethod;
-};
-const isShippingInfoReady = () => {
-    const address = storage.getItem('shipping_address');
-    return isAddressValid(address);
-};
-const isShippingMethodReady = checkout => !!checkout.shippingMethod;
 
 class CheckoutWrapper extends Component {
     static propTypes = {
         beginCheckout: func.isRequired,
+        cancelCheckout: func.isRequired,
         cart: shape({
             details: object,
             guestCartId: string,
             shippingMethods: array,
             totals: object
         }),
         checkout: shape({
+            billingAddress: object, // TODO: shape
```

Comment:
Resolve this and other TODOs? Or if we need to wait for something else can we open an issue to resolve?

Reply:
Oops, thanks

Reply:
Resolved by https://github.com/magento-research/pwa-studio/pull/799/commits/fb3ab5b146b877d59168c9d108a9adb38b48b982.